features/dd-sdk-android-profiling/src/test/kotlin/com/datadog/android/profiling/ProfilingTest.kt

```diff
@@ -0,0 +1,183 @@
+                profilerField.isAccessible = true
+                val profiler = profilerField.get(Profiling) as Profiler
+
+                assertThat(profiler).isNotInstanceOf(NoOpProfiler::class.java)
+                assertThat(profiler).isInstanceOf(PerfettoProfiler::class.java)
+            }
+    }
+
+    @Test
+    fun `M use PerfettoProfiler W enable called but start not called`() {
+        Mockito.mockStatic(ProfilingStorage::class.java)
+            .use {
+                // When
+                Profiling.enable(fakeConfiguration, mockSdkCore)
+
+                // Then
+                verify(mockSdkCore).registerFeature(any<ProfilingFeature>())
+
+                val profilerField = Profiling::class.java.getDeclaredField("profiler")
+                profilerField.isAccessible = true
+                val profiler = profilerField.get(Profiling) as Profiler
+
+                assertThat(profiler).isNotInstanceOf(NoOpProfiler::class.java)
+                assertThat(profiler).isInstanceOf(PerfettoProfiler::class.java)
+            }
+    }
+
+    @Test
+    fun `M use PerfettoProfiler W start called before enable`() {
+        // Given
+        val sdkInstanceNames = setOf(fakeInstanceName)
+
+        Mockito.mockStatic(ProfilingStorage::class.java).use {
+            // When
+            Profiling.start(mockContext, sdkInstanceNames)
+            Profiling.enable(fakeConfiguration, mockSdkCore)
+
+            // Then
+            verify(mockSdkCore).registerFeature(any<ProfilingFeature>())
+
+            val profilerField = Profiling::class.java.getDeclaredField("profiler")
+            profilerField.isAccessible = true
+            val profiler = profilerField.get(Profiling) as Profiler
+
+            assertThat(profiler).isNotInstanceOf(NoOpProfiler::class.java)
+            assertThat(profiler).isInstanceOf(PerfettoProfiler::class.java)
+        }
+    }
+
+    @Test
+    fun `M keep same profiler instance W start called multiple times`() {
+        // Given
+        val sdkInstanceNames = setOf(fakeInstanceName)
+
+        Mockito.mockStatic(ProfilingStorage::class.java)
+            .use {
+                // When
+                Profiling.start(mockContext, sdkInstanceNames)
+
+                val profilerField = Profiling::class.java.getDeclaredField("profiler")
+                profilerField.isAccessible = true
+                val firstProfiler = profilerField.get(Profiling) as Profiler
+
+                Profiling.start(mockContext, sdkInstanceNames)
+
+                val secondProfiler = profilerField.get(Profiling) as Profiler
+
+                // Then
+                assertThat(firstProfiler).isNotInstanceOf(NoOpProfiler::class.java)
+                assertThat(secondProfiler).isNotInstanceOf(NoOpProfiler::class.java)
+                assertThat(firstProfiler).isSameAs(secondProfiler)
+            }
+    }
+
+    private fun resetProfilerField() {
```

Comment:
same here, we can simply create `internal` method to reset the state if we want.